proposals/NNNN-atf-improvment-proposal.md

```diff
@@ -0,0 +1,62 @@
+# Improve Automated test framework
+
+* Proposal: [SDL-NNNN](NNNN-atf-improvment-proposal.md)
+* Author: [Alexander Kutsan](https://github.com/smartdevicelink)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / ATF]
+
+## Introduction
+
+The proposal is to make ATF more flexible, scalable and easy to use and integrate with popular CI systems.
+Also to make ATF reports easy to analyze and full of all required information. 
+
+## Motivation
+
+Current ATF requires and unfortunately allows changes during testing specific SDL scenarios. 
+Also it requires additional environment preparation and management during testing test sets
+and integrating ATF to popular CI systems.
+Need to make ATF more flexible and easy to use. And extend it scaling possibilities. 
+
+## Proposed solution
+
+There is 3 different layers of using ATF:
+
+1. Local executing and debugging concrete scenario even on modificated SDL and script
+2. Local test set execution, Logs, reports, traffic analyses. Set set can be easol local modificated and SDL also can be build customly
+3. CI execution of test set, collection Logs, reports, traffic logs for further analysis 
+
+ATF should be easy to use as for executing and debugging one concrate scenario on local workstation even with customly modified SDL.
+
+ATF architecture should be modified to support all this 3 layers of execution. Also ATF should support testing SDL on remote workstation/target. 
+
+
+
+## Detailed design
+
+Split ATF to parts:
+ * API Facade
+ * script runner
+ * Test set runner
+ * SDL Watchdog
+ * SDL Deployer 
+
+All this parts of ATF should be partially standalone applications.
+ATF should manage with it's with of git submodules system. 
+Such splitting will allow ATF easily change any part to one that matches environment according current  requirements (SDL on target for example)
+Also it allows to change some part of ATF to widely used open-source test frameworks (for example python-nose can be used ad Test set runner)
+Each part of ATF can be separately unit tested and keep own versioning and development cycle. 
+
+
+## Impact on existing code
+
+Mostly current ATF code will be keeped. But additional tools and infrastructure should be created.
+There will be a lot of minor changes in current ATF scripts, but they won't be rewrited from scratch. 
+
+
+## Alternatives considered
+
+It is possible to continue developing current approach :
+
+  Atf scripts repository contains bit amount of utils routine, scripts that required for CI activities and so on. 
+
+But It wont allow in futcher  testin remote SDL and it is rely not convien way to collect test reports, analyse SDL behaviour and keep versioning of scripts for SDL testing. 
```

Comment:
@LuxoftAKutsan "wont" -> "won't", "futcher" -> "further",  "testin" -> "testing", "convien" -> convenient. 
not+convenient -> inconvenient